### PR TITLE
enable ceph-mon.target service after monitor start, fix the ceph moni…

### PR DESCRIPTION
**Pull request by WingkaiHo**
Use ceph-ansbile install kraken, when reboot machine, the monitor service do not auto start,

By default:

$ systemctl is-enabled ceph-mon.target
disable

I found ceph-mon.target do not set enable, the monitor service do not auto start when machine reboot. So it used add the lines mentioned to fix this

### CODE REVIEW

**Comment by leseb** [comment]
I can reproduce. Normally the previous task should enable the service.
We also have these: https://github.com/ceph/ceph/blob/kraken/systemd/50-ceph.preset and they should enable the services. They might not be triggered.

@ktdreyer am I missing something here?

**Comment by leseb** [comment]
Package issue, fixing this on ceph

**Comment by leseb** [comment]
Fix proposed upstream: https://github.com/ceph/ceph/pull/14226
Please add a note into your patch to point to my fix, just a reminder so we can remove it once this is merged and backported.

Note this only affects Kraken.

**Comment by WingkaiHo** [comment]
Add note for patch.

File: roles/ceph-mon/tasks/start_monitor.yml [hunk 1]
```diff
@@ -5,3 +5,11 @@
     state: started
     enabled: yes
   changed_when: false
+
+# NOTE: This patch only effect Kraken. Fix when reboot machine, the monitor
```
Comment (leseb) [inline roles/ceph-mon/tasks/start_monitor.yml]: s/effect/affects/

File: roles/ceph-mon/tasks/start_monitor.yml [hunk 2]
```diff
@@ -5,3 +5,11 @@
     state: started
     enabled: yes
   changed_when: false
+
+# NOTE: This patch only effect Kraken. Fix when reboot machine, the monitor 
+# service do not auto start. Fix proposed upstream: ceph/ceph#14226
```
Comment (leseb) [inline roles/ceph-mon/tasks/start_monitor.yml]: "This fixes a bug when the monitor service does not start automatically after a reboot."

**Comment by leseb** [comment]
jenkins test ansible2.2-purge_cluster

File: roles/ceph-mon/tasks/start_monitor.yml (hunk 2 above)
Comment (WingkaiHo) [inline roles/ceph-mon/tasks/start_monitor.yml]: Update the note of pathc

**Comment by leseb** [comment]
@WingkaiHo thanks!

File: roles/ceph-mon/tasks/start_monitor.yml [hunk 3]
```diff
@@ -5,3 +5,11 @@
     state: started
     enabled: yes
   changed_when: false
+
+# NOTE: This patch only affects Kraken. This fixes a bug when the monitor service
```
Comment (leseb) [inline roles/ceph-mon/tasks/start_monitor.yml]: The CI is reporting an error on a second run, can you try this instead?

```
- name: enable the ceph-mon.target service
  systemd:
    name: ceph-mon.target
    enabled: yes
    masked: no
  changed_when: false
```

Comment (WingkaiHo) [inline roles/ceph-mon/tasks/start_monitor.yml]: @leseb  please try again.

**Comment by leseb** [comment]
Please do a single commit or another PR if you're not comfortable with git.
Thanks.

File: roles/ceph-mon/tasks/start_monitor.yml (hunk 3 above)
Comment (WingkaiHo) [inline roles/ceph-mon/tasks/start_monitor.yml]: @leseb , change, please try again.

**Comment by WingkaiHo** [comment]
OK

**Comment by leseb** [comment]
@WingkaiHo Thanks, once this is done I'll merge this.

**Comment by leseb** [comment]
Can you do this soon, we want to release stable-2.2 but the end of the week.
If you don't have time, I can do it for you.

**Comment by leseb** [comment]
jenkins test this please

**Comment by leseb** [comment]
done here: https://github.com/ceph/ceph-ansible/pull/1416

**Comment by WingkaiHo** [comment]
Thanks

**Pull request closed without merging**